walletrpc/service.proto

```diff
@@ -31,12 +31,31 @@ message TxFilter {
      bytes hash = 3;        // transaction ID (hash, txid)
 }
 
-// RawTransaction contains the complete transaction data. It also optionally includes 
-// the block height in which the transaction was included, or, when returned
-// by GetMempoolStream(), the latest block height.
```

Comment:
Bug: you're removing the documentation about its use in `GetMempoolStream` without either removing this usage from there, or fixing the bug that exists in `GetMempoolStream` (that it returns the chain tip height).

If we want consistency of how `RawTransaction` is used between `GetTransaction` and `GetMempoolStream`, we'd need to return `height = 0` (as sentinel for "in the mempool") from `GetMempoolStream`. However, given that downstream users currently expect `GetMempoolStream` to always return a height and would probably break badly if they got the genesis height, we should probably instead return the sentinel "chain tip height + 1" as the "mempool height". If we do this latter approach, then if we're still using `RawTransaction` (as we probably need to for protobuf compatibility) we will need to update this documentation to record the distinction.

Reply:
As we discussed, Zingo is currently the only project that is known to be using this height value, and they will update their usage. As such, I think that `chain_tip + 1` is not a suitable value for representing the mempool height, because the consumer of `RawTransaction` has no way to distinguish whether a returned transaction is in the mempool or has actually been mined; they would need to know how that `RawTransaction` value was retrieved, and pass that context along. As such, I will add a commit that fixes the `GetMempoolStream` behavior to return `0` for transactions in the mempool.

Reply:
Yeah, that is probably the safest change to make overall. It does however mean that there is a potential race condition where:
- Caller has no open `GetMempoolStream`.
- Caller checks the chain tip height.
- Chain tip height changes (in any direction), stopping any _existing_ `GetMempoolStream`
- Caller starts a _new_ `GetMempoolStream`

and the caller will think the mempool is associated with a target height that it is not.

Reply:
The core issue is that `GetMempoolStream` is not a great API (in particular because it streams the entire mempool from scratch as each block comes in), and we should plan to replace it with a better API (that provides a mempool diff along with the height transition associated with it).

Reply:
Fixed in b5f3f6348df4f6f4a2690a8f8806c571557fe642